src/util/misc.ts

```diff
@@ -285,3 +285,8 @@ export function updateMap<T1, T2>(
         updatee.set(key, val);
     }
 }
+
+export function last<T>(arr: T[]): T {
+    assert(arr.length > 0, `Internal Error: Unexpected empty array`);
```

Comment:
I guess we better warn developer(s) that `arr` s expected to be non-empty (`length > 0`). I generally like the idea of shorthand, however I think that it sometimes may be  better to delegate error handling a logic at particular callsite (because of possible edge cases) without using `try`/`catch` or workarounds.

Now I would suggest something that I would not do quite often: add a boolean flag `throwOnEmpty = true` and consider overloading.
```typescript
export function last<T>(arr: T[]): T;
export function last<T>(arr: T[], throwOnEmpty: true): T;
export function last<T>(arr: T[], throwOnEmpty: false): T | undefined;
export function last<T>(arr: T[], throwOnEmpty = true): T | undefined {
    assert(!throwOnEmpty || arr.length > 0, `Internal Error: Unexpected empty array`);

    return arr[arr.length - 1];
}
```
We can also think if the flag is `allowEmpty` instead if you prefer, but idea is to derive the checks to the type system and not cause confusion / accidental bugs. Other thing would be to have two different implementations (one with `assert()` call and one without it).

Reply:
Fixed in https://github.com/ConsenSys/scribble/pull/41/commits/7dfdbafe0d9b836e1e7ab083c07dc4eb3bdfa24c